pkg/internal/approver/validation/validator_test.go

```diff
@@ -91,3 +95,48 @@ func newCertificateRequest(namespace string) cmapi.CertificateRequest {
 	request.SetNamespace(namespace)
 	return request
 }
+
+func Test_Validator_Validate_ServiceAccount(t *testing.T) {
+	v := &validator{expression: "ServiceAccount(cr.username).isServiceAccount() && self.startsWith('spiffe://acme.com/ns/%s/sa/%s'.format([ServiceAccount(cr.username).getNamespace(),ServiceAccount(cr.username).getName()]))"}
```

Comment:
```suggestion
	v := &validator{expression: "isServiceAccount(cr.username) && self.startsWith('spiffe://acme.com/ns/%s/sa/%s'.format([serviceAccount(cr.username).getNamespace(),serviceAccount(cr.username).getName()]))"}
```

I would suggest a slightly modified API to this new feature. Please let me know what you think and if it is possible to implement. I will await a review of the code/implementation until we are aligned on how this will will appear for the user. 😺

Reply:
Added you changes and modified the tests- I also havent had a chance to test this in a real cluster yet